apps/protocol-frontend/src/components/ContributionTypesTable.tsx

```diff
@@ -1,52 +1,54 @@
 import { useMemo } from 'react';
-import { format } from 'date-fns';
-import { Box, Table, Tbody, Td, Th, Thead, Tr, chakra } from '@chakra-ui/react';
-import { IoArrowDown, IoArrowUp } from 'react-icons/io5';
-import { useTable, useSortBy } from 'react-table';
 import { isAfter } from 'date-fns';
+import { Box, chakra, Table, Tbody, Td, Th, Thead, Tr } from '@chakra-ui/react';
+import { IoArrowDown, IoArrowUp } from 'react-icons/io5';
+import { useSortBy, useTable } from 'react-table';
+import { ContributionItem } from '@govrn/protocol-client';
 
-const ContributionTypesTable = ({ contributionTypesData }: any) => {
+const ContributionTypesTable = ({
+  contributionTypesData,
+}: {
+  contributionTypesData: ContributionItem[];
+}) => {
   const uniqueKey = 'name';
 
-  const uniqueContributions = [
+  console.dir(contributionTypesData);
+
+  // FIXME: This will be re-calculated for each render.
+  const uniqueContributions: ContributionItem[] = [
     ...new Map(
       contributionTypesData
-        .sort((firstContribution: any, nextContribution: any) =>
+        .sort((firstContribution, nextContribution) =>
           isAfter(
-            new Date(firstContribution.engagementDate),
-            new Date(nextContribution.engagementDate)
+            new Date(firstContribution.date_of_engagement),
+            new Date(nextContribution.date_of_engagement)
           )
             ? 1
             : -1
         )
-        .map((contributionType: any) => [
+        .map((contributionType) => [
           contributionType.activity_type[uniqueKey],
           contributionType,
         ])
     ).values(),
   ];
 
+  // FIXME: divide these into different states.
   const data = useMemo(
     () =>
-      uniqueContributions.map((contributionType: any) => ({
+      uniqueContributions.map((contributionType) => ({
         id: contributionType.id,
         name: contributionType.name,
         total: contributionTypesData.filter(
-          (contribution: any) =>
+          (contribution) =>
             contribution.activity_type.name ===
             contributionType.activity_type.name
         ).length,
-        lastOccurrence: contributionType.engagementDate,
+        lastOccurrence: contributionType.date_of_engagement,
         activityType: contributionType.activity_type.name,
-        guilds: contributionType.guild,
-        submissionDate: format(
-          new Date(contributionType.date_of_submission),
-          'P'
-        ),
```

Comment:
What is the difference between using this formatting and consuming the raw string

Reply:
database format: `2022-07-01T04:00:00.000Z`. 
After formatting: `07/19/2022`